src/wasm-lib/kcl/src/execution/artifact.rs

```diff
@@ -794,13 +812,48 @@ fn artifacts_to_update(
                         source_ranges: vec![range],
                     })
                 })?;
-                return_arr.push(Artifact::Wall(Wall {
+                let extra_artifact = _exec_artifacts.values().find_map(|a| {
+                    if let Artifact::StartSketchOnFace { face_id: id, .. } = a {
+                        if *id == face_id.0 {
+                            return Some(a.clone());
+                        }
+                    }
+                    None
+                });
+                let sketch_on_face_source_range = extra_artifact.and_then(|a| match a {
+                    Artifact::StartSketchOnFace { source_range, .. } => Some(source_range),
+                    _ => None,
+                });
+                let mut wall_artifact = Wall {
+                    id: face_id,
+                    seg_id: curve_id,
+                    edge_cut_edge_ids: Vec::new(),
+                    sweep_id: path.sweep_id.expect("Expected sweep_id to be Some"),
+                    path_ids: Vec::new(),
+                    face_code_ref: CodeRef {
+                        range,
+                        path_to_node: path_to_node.clone(),
+                    },
+                };
+
+                if let Some(sketch_on_face_source_range) = sketch_on_face_source_range {
+                    wall_artifact.face_code_ref = CodeRef {
+                        range: sketch_on_face_source_range,
+                        path_to_node: path_to_node.clone(),
+                    };
+                }
```

Comment:
`wall_artifact` is getting built with the new sketch-on-face source range here, but it's getting thrown away. Fix incoming.

Reply:
Fixed in 11eceefedf2e166b0ec69875e3de417afa93517a.

Reply:
Dope! You might have fixed my point 4 [here](https://github.com/KittyCAD/modeling-app/pull/5196#pullrequestreview-2606553771), let me test it